cs/src/Connections/TunnelRelayTunnelHost.cs

```diff
@@ -672,25 +605,20 @@ private void OnSshChannelOpening(object? sender, SshChannelOpeningEventArgs e)
     /// <inheritdoc />
     public override async Task RefreshPortsAsync(CancellationToken cancellation)
     {
-        if (Tunnel == null || ManagementClient == null)
+        if (! await RefreshTunnelAsync(includePorts: true, cancellation))
         {
             return;
         }
 
-        var updatedTunnel = await ManagementClient.GetTunnelAsync(
-            Tunnel, new TunnelRequestOptions { IncludePorts = true });
-
-        var updatedPorts = updatedTunnel?.Ports ?? Array.Empty<TunnelPort>();
-        Tunnel.Ports = updatedPorts;
-
+        var updatedPorts = Tunnel?.Ports ?? Array.Empty<TunnelPort>();
```

Comment:
If Tunnel.ports is empty or null can we just return immediately

Reply:
We also stop forwarding from the ports that are now missing on line 639.